lib/util/zip.ex

```diff
@@ -0,0 +1,143 @@
+# Copyright(c) 2015-2019 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule Antikythera.Zip do
+  @moduledoc """
+  Wrapper module for `zip` command.
+
+  For the consistency in working with antikythera and other gears, scope of this module is limited under a temporary directory reserved by `Antikythera.Tmpdir.make/2`.
+
+  Functions only accept absolute paths for both source and resulting archive.
+  """
+
+  alias Croma.Result, as: R
+  alias Antikythera.Context
+  alias Antikythera.ExecutorPool.Id, as: EPoolId
+  alias AntikytheraCore.TmpdirTracker
+
+  @typep opts :: {:encryption, boolean} | {:password, String.t}
+
+  defmodule FileName do
+    use Croma.SubtypeOfString, pattern: ~r/^(?!.*\/\.{0,2}\z).*\z/
+  end
+
+  @doc """
+  Creates a ZIP file.
+
+  Encryption using a password is supported in which `encryption` option is set `true`.
+
+  ## Example
+    Tmpdir.make(context, fn tmpdir ->
+      src_path = tmpdir <> "/src.txt"
+      zip_path = tmpdir <> "/archive.zip"
+      File.write!(src_path, "text")
+      Antikythera.Zip.zip(context, zip_path, src_path, [encryption: true, password: "password"])
+      |> case do
+        {:ok, archive_path} ->
+          ...
+      end
+      ...
+    end)
+  """
+  defun zip(
+    context_or_epool_id :: v[EPoolId.t | Context.t],
+    zip_raw_path        :: v[FileName.t],
+    src_raw_path        :: v[String.t],
+    opts                :: v[list(opts)] \\ []
+  ) :: R.t(Path.t) do
```

Comment:
As it is written as follows in [Antikythera Style Guide](https://hexdocs.pm/antikythera/style_guide.html), please be aware of the style unification of the existing codes.
> 2. Be consistent with surrounding code.

The following format is generally used within Antikythera.
```ex
defun zip(context_or_epool_id :: v[EPoolId.t | Context.t],
          zip_raw_path        :: v[FileName.t],
          src_raw_path        :: v[String.t],
          opts                :: v[list(opts)] \\ []) :: R.t(Path.t) do

Reply:
Fixed in 2a164a4 .